experimental/devices/cap1188/cap1188.go

```diff
@@ -0,0 +1,554 @@
+		// the polling cycle.
+		// default 0: The device will always power down as much as possible
+		// during the time between the end of the last conversion and the
+		// end of the polling cycle.
+		byte(1)<<5 |
+		//  Determines whether the LED Mirror Control register bits are
+		//  linked to the LED Polarity bits. Setting this bit blocks the
+		//  normal behavior which is to automatically set and clear the LED
+		//  Mirror Control bits when the LED Polarity bits are set or
+		//  cleared.
+		//  default 0: When the LED Polarity controls are set, the
+		//  corresponding LED Mirror control is automatically set. Likewise,
+		//  when the LED Polarity controls are cleared, the corresponding
+		//  LED Mirror control is also cleared.
+		byte(0)<<4 |
+		// Determines whether the Noise Status bits will show RF Noise as
+		// the only input source.
+		// default 0: The Noise Status registers will show both RF noise and
+		// low frequency EMI noise if either is detected on a capacitive
+		// touch sensor input.
+		byte(0)<<3 |
+		// Determines whether the RF noise filter is enabled.
+		// default 0:  If RF noise is detected by the analog block, the
+		// delta count on the corresponding channel is set to 0. Note that
+		// this does not require that Noise Status bits be set.
+		byte(0)<<2 |
+		byte(0)<<1 |
+		// Controls the interrupt behavior when a release is detected on a button.
+		// when 0: An interrupt is generated when a press is detected and
+		// again when a release is detected and at the repeat rate (if enabled)
+		// when 1:  An interrupt is generated when a press is detected and
+		// at the repeat rate but not when a release is detected.
+		intOnRel<<0)
+	if Debug {
+		fmt.Printf("cap1188: Config2 mask: %08b\n", config2)
+	}
+	if err = d.regWrapper.WriteUint8(0x44, config2); err != nil {
+		return fmt.Errorf("failed to set the device configuration 2 - %s", err)
+	}
+
+	return nil
+}
+
+// setBit sets a specific bit on a register
+// TODO: avoid reading before writing, keep states in memory
+func (d *Dev) setBit(regID uint8, idx int) error {
+	v, err := d.regWrapper.ReadUint8(regID)
+	if err != nil {
+		return err
+	}
+	v |= (1 << uint8(idx))
+	return d.regWrapper.WriteUint8(regID, v)
+}
+
+// clearBit clears a specific bit on a register
+// TODO: avoid reading before writing, keep states in memory
+func (d *Dev) clearBit(regID uint8, idx int) error {
+	v, err := d.regWrapper.ReadUint8(regID)
+	if err != nil {
+		return err
+	}
+	v &= ^(1 << uint8(idx))
+	return d.regWrapper.WriteUint8(regID, v)
+}
+
+// resetMemInputStatus resets the in memory slice containing the state of the
+// inputs.
+func resetMemInputStatus() {
+	inputStatuses = make([]TouchStatus, nbrOfLEDs)
+}
+
+// b is the byte to check and position is the bit position
+// index 0 where 7 is the "most left bit".
+func isBitSet(b byte, pos uint8) bool {
+	return (b>>pos)&1 == 1
+}
+
+// twoComplementsToInt returns the int value of a standard 2’s complement number
```

Comment:
Same,  it's not complex enough to warrant extracting.

Reply:
while this isn't that complex nor used in multiple places, this data is formatted in a special way and I find that isolating the logic makes that clear. That said, after looking at the datasheet, this approach isn't used anywhere else so I'll indeed inline the decoding.

Reply:
that said @maruel should I also take the same approach for `isBitSet()`, it's pretty simple logic and only has 1 call site?

Reply:
Yes, thanks.